### PR TITLE
Scan number

### PR DESCRIPTION
**Pull request by jeffhappily**

### CODE REVIEW

File: src/compiler/scanner/index.ts [hunk 1]
```diff
@@ -172,6 +172,36 @@ export class Scanner {
           debug(`<char literal isClosed: ${isCharLiteralClosed}>`)
           return SyntaxSet.CharLiteral
 
+        case CharacterCodes._1:
```
Comment (zypeh) [inline src/compiler/scanner/index.ts:175]: Seems like we didn't parse the character `0` ?

**Review by zypeh — CHANGES_REQUESTED** [review]

File: src/compiler/scanner/index.ts (hunk 1 above)
Comment (jeffhappily) [reply src/compiler/scanner/index.ts:175]: I thought we should separate out number that starts with zero, like octal or hex?

Comment (zypeh) [reply src/compiler/scanner/index.ts:175]: 👍 fair enough

File: src/compiler/scanner/index.ts [hunk 2]
```diff
@@ -181,7 +181,8 @@ export class Scanner {
         case CharacterCodes._7:
         case CharacterCodes._8:
         case CharacterCodes._9:
-          let isFloat: boolean = false
+        case CharacterCodes.dot:
```
Comment (zypeh) [inline src/compiler/scanner/index.ts:184]: I don't think we really need starting dot for floating numbering... Any practical usage or strong reason?

**Pull request merged**